sr_port/restrict.c

```diff
@@ -68,31 +68,32 @@ error_def(ERR_TEXT);
 
 #define	PUT_FLNAME_IN_MAPPING_FILE(RPATH, FPATH, FP, C_CALL_NAME, M_REF_NAME, STAT_RM, SAVE_ERRNO, ERR_STR)	\
 {														\
-	if (!ACCESS(FPATH,F_OK))		/*Filter file exists, now check modified time*/			\
+	if (!ACCESS(FPATH, F_OK))		/* Filter file exists, now check modified time */		\
 	{													\
 		Stat(RPATH, &rTime);										\
 		rmtime = rTime.st_mtim;										\
 		Stat(FPATH, &fTime);										\
 		fmtime = fTime.st_mtim;										\
-		if ((rmtime.tv_sec > fmtime.tv_sec) || (rmtime.tv_nsec >= fmtime.tv_nsec))			\
-		{	/*Delete the older mapping file and recreate new if required*/				\
+		/* Check if restrict.txt file modification time (rmtime) is newer than				\
+		 * filter_commands.tab file modificatoin time (fmtime). If so, recreate filter_commands.tab.	\
+		 */												\
+		if ((rmtime.tv_sec > fmtime.tv_sec)								\
+			|| (rmtime.tv_sec == fmtime.tv_sec) && (rmtime.tv_nsec >= fmtime.tv_nsec))		\
```

Comment:
Maybe add another set of parens to group the ANDed condition separately from the ORed condition?